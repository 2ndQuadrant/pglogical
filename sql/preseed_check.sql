-- Verify data from preseed.sql has correctly been cloned
SELECT * FROM pglogical_regress_variables()
\gset

\c :provider_dsn
\d some_local_tbl
SELECT * FROM some_local_tbl ORDER BY id;

\d some_local_tbl1
SELECT * FROM some_local_tbl1 ORDER BY id;

\d some_local_tbl2
SELECT * FROM some_local_tbl2 ORDER BY id;

\d some_local_tbl3
SELECT * FROM some_local_tbl3 ORDER BY id;

\c :subscriber_dsn

\d some_local_tbl
SELECT * FROM some_local_tbl ORDER BY id;

\d some_local_tbl1
SELECT * FROM some_local_tbl1 ORDER BY id;

\d some_local_tbl2
SELECT * FROM some_local_tbl2 ORDER BY id;

\d some_local_tbl3
SELECT * FROM some_local_tbl3 ORDER BY id;

\c :provider_dsn
\set VERBOSITY terse
SELECT pglogical.replicate_ddl_command($$
        DROP TABLE public.some_local_tbl;
        DROP TABLE public.some_local_tbl1;
        DROP TABLE public.some_local_tbl2;
        DROP TABLE public.some_local_tbl3;
$$);
