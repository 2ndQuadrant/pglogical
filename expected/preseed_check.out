-- Verify data from preseed.sql has correctly been cloned
SELECT * FROM pglogical_regress_variables()
\gset
\c :provider_dsn
\d some_local_tbl
                         Table "public.some_local_tbl"
 Column |  Type   |                          Modifiers                          
--------+---------+-------------------------------------------------------------
 id     | integer | not null default nextval('some_local_tbl_id_seq'::regclass)
 key    | text    | not null
 data   | text    | 
Indexes:
    "some_local_tbl_pkey" PRIMARY KEY, btree (id)
    "some_local_tbl_key_key" UNIQUE CONSTRAINT, btree (key)

SELECT * FROM some_local_tbl ORDER BY id;
 id | key  | data  
----+------+-------
  1 | key1 | data1
  2 | key2 | 
  3 | key3 | data3
(3 rows)

\d some_local_tbl1
                         Table "public.some_local_tbl1"
 Column |  Type   |                          Modifiers                           
--------+---------+--------------------------------------------------------------
 id     | integer | not null default nextval('some_local_tbl1_id_seq'::regclass)
 key    | text    | not null
 data   | text    | 
Indexes:
    "some_local_tbl1_key_key" UNIQUE CONSTRAINT, btree (key)

SELECT * FROM some_local_tbl1 ORDER BY id;
 id | key  | data  
----+------+-------
  1 | key1 | data1
  2 | key2 | 
  3 | key3 | data3
(3 rows)

\d some_local_tbl2
                         Table "public.some_local_tbl2"
 Column |  Type   |                          Modifiers                           
--------+---------+--------------------------------------------------------------
 id     | integer | not null default nextval('some_local_tbl2_id_seq'::regclass)
 key    | text    | 
 data   | text    | 

SELECT * FROM some_local_tbl2 ORDER BY id;
 id | key  | data  
----+------+-------
  1 | key1 | data1
  2 | key2 | 
  3 | key3 | data3
(3 rows)

\d some_local_tbl3
Table "public.some_local_tbl3"
 Column |  Type   | Modifiers 
--------+---------+-----------
 id     | integer | 
 key    | text    | 
 data   | text    | 

SELECT * FROM some_local_tbl3 ORDER BY id;
 id | key  | data  
----+------+-------
    | key1 | data1
    | key2 | 
    | key3 | data3
(3 rows)

\c :subscriber_dsn
\d some_local_tbl
                         Table "public.some_local_tbl"
 Column |  Type   |                          Modifiers                          
--------+---------+-------------------------------------------------------------
 id     | integer | not null default nextval('some_local_tbl_id_seq'::regclass)
 key    | text    | not null
 data   | text    | 
Indexes:
    "some_local_tbl_pkey" PRIMARY KEY, btree (id)
    "some_local_tbl_key_key" UNIQUE CONSTRAINT, btree (key)

SELECT * FROM some_local_tbl ORDER BY id;
 id | key | data 
----+-----+------
(0 rows)

\d some_local_tbl1
                         Table "public.some_local_tbl1"
 Column |  Type   |                          Modifiers                           
--------+---------+--------------------------------------------------------------
 id     | integer | not null default nextval('some_local_tbl1_id_seq'::regclass)
 key    | text    | not null
 data   | text    | 
Indexes:
    "some_local_tbl1_key_key" UNIQUE CONSTRAINT, btree (key)

SELECT * FROM some_local_tbl1 ORDER BY id;
 id | key | data 
----+-----+------
(0 rows)

\d some_local_tbl2
                         Table "public.some_local_tbl2"
 Column |  Type   |                          Modifiers                           
--------+---------+--------------------------------------------------------------
 id     | integer | not null default nextval('some_local_tbl2_id_seq'::regclass)
 key    | text    | 
 data   | text    | 

SELECT * FROM some_local_tbl2 ORDER BY id;
 id | key | data 
----+-----+------
(0 rows)

\d some_local_tbl3
Table "public.some_local_tbl3"
 Column |  Type   | Modifiers 
--------+---------+-----------
 id     | integer | 
 key    | text    | 
 data   | text    | 

SELECT * FROM some_local_tbl3 ORDER BY id;
 id | key | data 
----+-----+------
(0 rows)

\c :provider_dsn
\set VERBOSITY terse
SELECT pglogical.replicate_ddl_command($$
        DROP TABLE public.some_local_tbl;
        DROP TABLE public.some_local_tbl1;
        DROP TABLE public.some_local_tbl2;
        DROP TABLE public.some_local_tbl3;
$$);
 replicate_ddl_command 
-----------------------
 t
(1 row)

